Pin python2-psycopg2 package

sbesson opened the pull request:
See https://www.postgresql.org/message-id/EA7F5495-64AE-4BEF-81B1-7FA921BBE690%40gmail.com

The latest versions of `python2-psycopg2` pushed to the PostgreSQL YUM repo coupled with the obsolescence of `python-psycopg2` have broken compatibility with PSQL 9.4, 9.5 and 9.6. This PR
caps the version of psycopg2 installed to fix the `postgresql_user` module.

Additionally, this PR makes two changes:
- use `centos:7` as the `image` in `molecule.yml` since `centos:latest` is now `centos:8` and `image_version` is dropped in Molecule 2
- updates `vars/main.yml` to use the single RedHat Yum repository - see https://www.postgresql.org/message-id/6f1e601300d575195d4f0d8a066ef4abf4c90c99.camel@gunduz.org 

Given the impact on all playbooks both for fresh deployments and production deployments, proposing to get this released as an emergency 3.3.1 /cc @joshmoore @manics @jburel

sbesson committed:
Update postgresql_repourl to use the single distribution PosgreSQL RPM

See https://www.postgresql.org/message-id/6f1e601300d575195d4f0d8a066ef4abf4c90c99.camel@gunduz.org for reference
The separate RPMs per major PostgreSQL versions have been deprecated in favor
of a single distribution RPM.

manics commented:
To be tested as an upgrade against pub-omero before releasing, but looks fine

sbesson committed:
Combine image name and tag

sbesson commented:
The current state of the role suffers from two distinct issues. Trying to summarize here:

- as of yesterday, the Docker `centos:latest` image points to `centos:8`. In theory this should not affect any of our roles which are tested against `centos:7`. However I realized that on Molecule 2, the `image_version` field has been silently dropped. 4357147d5190786b27aae164c172b7fec7ff17c6 addresses this but PRs with similar changes  will need to be opened against all roles upgraded to Molecule 2 to prevent the image creation step from failing

- more critical, the initial problem that this PR tried to address seems to arise from the fact that the `python-psycopg2` (required for Ansible's core `postgresql_user` module) is now obsolete in favor of `python2-psycopg2`. The latest upstream releases of this package seem to have incompatibilities with versions of PostgreSQL 9.6 and earlier [as reported here](https://www.postgresql.org/message-id/EA7F5495-64AE-4BEF-81B1-7FA921BBE690%40gmail.com).

manics commented:
Installing the previous release works:
`yum install -y postgresql96-server python2-psycopg2-2.8.3-1.rhel7.x86_64` or `yum downgrade python2-psycopg2-2.8.3-1.rhel7.x86_64`

sbesson committed:
Pin python2-psycopg2 to 2.8.3-1

As of today, this is the latest version compatible with all supported PSQL
versions - see https://www.postgresql.org/message-id/EA7F5495-64AE-4BEF-81B1-7FA921BBE690%40gmail.com for the upstream report

sbesson committed:
Try uncapping psycopg2 but forcing the repoid

See https://www.postgresql.org/message-id/9a97200b9bc167297e88f7be482634ddfb340ccf.camel%40gunduz.org
